### ISSUE TITLE
clipnorm or global_clipnorm

### ISSUE THREAD

**Issue by RoyStegeman** [NNPDF/nnpdf#1210]
Since the 2.4.0 release of tf, optimizers now also support `global_clipnorm`, see the [tensorflow docs](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer#attributes_1). The difference between the regular clipnorm and global_clipnorm, is that the regular clipnorm clips individual gradients if their L2-norm is larger than the clipnorm value, whereas global glipnorm rescales all gradients if the global L2-norm of the gradients is larger than the global_clipnorm value. 

Originally I intended to use global_clipnorm since that does not change the path taken by the optimizer, but really only prevents 'exploding' gradients, whereas of course rescaling only some gradients does change the path taken by the optimizer. However, when I first looked into ways to prevent the exploding gradients global_clipnorm was not yet supported, and regular clipnorm seemed to do the job just fine. 

I don't think it will affect the resulting PDFs very much (a [comparison between the baseline fit with clipnorm, and a new fit with global_clipnorm](https://vp.nnpdf.science/A-p56hWpSlOIshGt6Mahng==/) does not show much of a difference) but perhaps it's something worth checking if, because of the changes to training fraction and dataset, we decide to do  a new hyperopt at some point.

-----------

**Edit:** I ran a hyperopt with the global_clipnorm, and though the best setup obtained a slightly lower chi2 than what was found with regular clipnorm, this difference is at the level of statistical fluctuations. So no significant improvement for global_clipnrom with respect to the regular clipnorm was observed.

**Comment by scarlehoff** [comment]
@RoyStegeman what is the status of this?

**Comment by RoyStegeman** [comment]
I don't care about this anymore
